OpenPanelSDK/src/main/java/com/dev/openpanelsdk/OpenPanel.kt

```diff
@@ -238,23 +204,177 @@ open class OpenPanel(protected val options: Options) {
         currentQueue.forEach { send(it) }
     }
 
-    open fun setupAutomaticTracking() {
-        // Base implementation (empty)
+    private fun getDefaultEventProperties(): Map<String, Any> {
```

Comment:
Would want to avoid adding all these properties here. Most of them could be constructed into a user-agent. 

Cant we just use `String ua=new WebView(this).getSettings().getUserAgentString();` or `String userAgent = System.getProperty("http.agent");`?

Reply:
These properties could be removed `os`, `os_version`, `manufacturer`, `brand` and `model`

Reply:
And all other properties, can we prefix them with `__` (2 underlines) it means its a protected property sent by the SDK.

Will be easier to filter out what properties the user has sent vs the SDK

Reply:
Would also want to keep the properties `camelCase`

Reply:
> These properties could be removed `os`, `os_version`, `manufacturer`, `brand` and `model`

I knew that some fields are already defined like os, os_version ... etc as there was slots for all this values in open panel admin panel but I was not able to find how to populate from the api, I have also reviewed the docs for OpenPanel api but there it's also not defined on how to populate this values.

Can you send me the request parameter from which I can populate this values direct in defined values and not in properties.

Reply:
Comes from the user-agent:  `os`, `os_version`, `manufacturer`, `brand` and `model`
Any other value should just be `__xxx`

So the request would look like this:

```
curl 'https://api.openpanel.dev/track' \
  -H 'content-type: application/json' \
  -H 'openpanel-client-id: xxx' \
  -H 'openpanel-client-secret: xxx' \
  -H 'openpanel-sdk-name: kotlin' \
  -H 'openpanel-sdk-version: 0.0.1' \
  -H 'user-agent: Dalvik/2.1.0 (Linux; U; Android 14; SM-M536B Build/UP1A.231005.007)' \
  --data-raw '{"type":"track","payload":{"name":"the_event_name","properties":{"__buildVersion":"1.2.3", "__version":"92"}}}'
```

**Other properties**

For the other properties you send, bluetooht, nfc, wifi etc. I don't think this is information I want to auto track. This is things that I think the end-user will have to add as global properties.

Reply:
> Would want to avoid adding all these properties here. Most of them could be constructed into a user-agent.
> 
> Cant we just use `String ua=new WebView(this).getSettings().getUserAgentString();` or `String userAgent = System.getProperty("http.agent");`?

When I use *WebView(this).getSettings().getUserAgentString()* then it give this result for user
![image](https://github.com/user-attachments/assets/646659a4-5d1e-4ae1-84a6-f6344c37d0ef)
![image](https://github.com/user-attachments/assets/2ecdec13-6a38-4cea-8217-c71ce6a2a4c7)

And When I use *System.getProperty("http.agent");* then it gives 

![image](https://github.com/user-attachments/assets/62a4d6aa-2548-4c41-97ed-f030c1d30b1c)


My device is a OnePlus device but the device it shows in dashboard is Oppo, how are you fetching manufacturer from user-agent ?

Reply:
And why we are considering browser here, I think showing browser does not make sense in an android app as it an individual and not a webview!

Reply:
> **Other properties**
> 
> For the other properties you send, bluetooht, nfc, wifi etc. I don't think this is information I want to auto track. This is things that I think the end-user will have to add as global properties.

I have added all the properties same as mix panel, MixPanel also auto track this properties, I think keeping this will help some dev and clients to get this extra data without any extra code, if we keep this as global then devs have to fetch this from code and then send it in global properties

Reply:
> My device is a OnePlus device but the device it shows in dashboard is Oppo, how are you fetching manufacturer from user-agent? 

I use a user-agent library (`ua-parser-js`) to parse the user-agent. Looking at what user-agent you send `Dalvik/2.1.0 (Linux; U; Android 14; CPH2487 Build/UKQ1.231108.001)` and `Mozilla/5.0 (Linux; Android 14; CPH2487 Build/UKQ1.231108.001; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/130.0.6723.86 Mobile Safari/537.36` it does not really matter what you pic. Sadly this has not all info I would expect.

Maybe I should support a `__brand` (`OnePlus`), `__model` (`CPH2487`) as properties (so we can override the default behaviour. 

Would be interesting to know what value you get if you visit this site: https://www.whatismybrowser.com/detect/what-is-my-user-agent/ (from your OnePlus phone)

> And why we are considering browser here, I think showing browser does not make sense in an android app as it an individual and not a webview!

This is mainly a web feature so I agree it don't need to show it here. But it will rather be a feature to be able to customize what columns it will show.

Reply:
> Would be interesting to know what value you get if you visit this site: https://www.whatismybrowser.com/detect/what-is-my-user-agent/ (from your OnePlus phone)

Visiting the link from my phone gives me *Mozilla/5.0 (Linux; Android 10; K) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/129.0.0.0 Mobile Safari/537.36* which is wrong info as the device is running Android 14!